modules/artifact/src/test/java/org/mule/runtime/module/artifact/classloader/IBMMQResourceReleaserTestCase.java

```diff
@@ -179,6 +183,12 @@ public void mBeansTest() throws Exception {
     assertThat(countMBeans(artifactClassLoader), is(0));
   }
 
+  @Test
```

Comment:
Please also add a test for the scenario reported by the customer (with the driver declared in a shared domain classloader).

And also a test in which there are different applications, each with its driver and only one is undeployed and the other must not be affected.